### DISCUSSION TITLE
Failed to load PostCSS config

### DISCUSSION THREAD

**Discussion by ahosker** [tailwindlabs/tailwindcss#11199] [Help]
**MISC**
This could be a bug(s) in Laravel Sail, Livewire or Tailwind CSS.

**What version of Tailwind CSS are you using?**

v3.3.2

**What build tool (or framework if it abstracts the build tool) are you using?**
Vite

**What version of Node.js are you using?**

v18.16.0

**What operating system are you using?**

Using Laravel Sail on Ubuntu

**Describe your issue**

I installed ubuntu server.
I updated ubuntu server.
I used Laravel Sail Linux Guide ( https://laravel.com/docs/10.x )
I installed Tall Stack ( https://tallstack.dev/ )

```
$ sail npm run dev

> dev
> vite


  VITE v4.3.5  ready in 556 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: http://172.19.0.7:5173/
  ➜  press h to show help

  LARAVEL v10.10.0  plugin v

  ➜  APP_URL: http://localhost
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[Failed to load PostCSS config: Failed to load PostCSS config (searchPath: /var/www/html): [Failed to load PostCSS config] Failed to load PostCSS config (searchPath: /var/www/html): [ReferenceError] module is not defined in ES module scope
This file is being treated as an ES module because it has a '.js' file extension and '/var/www/html/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
ReferenceError: module is not defined in ES module scope
This file is being treated as an ES module because it has a '.js' file extension and '/var/www/html/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
    at file:///var/www/html/postcss.config.js:1:1
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
]

Node.js v18.16.0
```

The following command fixes the issue:

`mv postcss.config.js postcss.config.cjs`

Though this shouldn't be required? 

After that fix, running NPM run dev I get the error:
```
8:06:19 PM [vite] Internal server error: __dirname is not defined
      at file:///var/www/html/node_modules/laravel-vite-plugin/dist/index.mjs:126:39
      at call (file:///var/www/html/node_modules/vite/dist/node/chunks/dep-934dbc7c.js:49019:7)
      at next (file:///var/www/html/node_modules/vite/dist/node/chunks/dep-934dbc7c.js:48963:5)
      at file:///var/www/html/node_modules/vite/dist/node/chunks/dep-934dbc7c.js:61287:17
      at viteHtmlFallbackMiddleware (file:///var/www/html/node_modules/vite/dist/node/chunks/dep-934dbc7c.js:61367:16)
      at call (file:///var/www/html/node_modules/vite/dist/node/chunks/dep-934dbc7c.js:49019:7)
      at next (file:///var/www/html/node_modules/vite/dist/node/chunks/dep-934dbc7c.js:48963:5)
      at viteServeStaticMiddleware (file:///var/www/html/node_modules/vite/dist/node/chunks/dep-934dbc7c.js:46030:20)
      at call (file:///var/www/html/node_modules/vite/dist/node/chunks/dep-934dbc7c.js:49019:7)
      at next (file:///var/www/html/node_modules/vite/dist/node/chunks/dep-934dbc7c.js:48963:5)
```

**Comment by ahosker** [comment]
I'd love to get this solved (selfishly) but because I'm using the default Laravel Development Installation Guide and unmodified TailwindCSS, Alpine & Livewire install. This is a friction to newbies getting started, we all want **Laravel to work out of the box** (especially with the awesome TALL stack).

I'd look into fixing but unfortunately, I don't know where to start. Is it Laravel Sail (@driesvints), Alpine/Livewire (@calebporzio), or TailwindCSS (@adamwathan).

**My understanding so far:**
The **PostCSS** config is 'fixable' with a rename but needs a change in the way NPM or Vite is setup. (not my area).

The "**__dirname**" error, I have read is due to Laravel Sail using 'stable' Node and NPM. Updating the Docker File to use Node 20 fixes this issue.

```
    laravel.test:
        build:
            context: ./vendor/laravel/sail/runtimes/8.2
            dockerfile: Dockerfile
            args:
                WWWGROUP: '${WWWGROUP}'
                #NODE_VERSION: '16' #default
                NODE_VERSION: '20' #working
```

**Comment by ahosker** [comment]
When following instructions and doing it on Windows (instead of a remote Linux machine)
I do get the PostCSS config error but not the __dirname error.

**Comment by ahosker** [comment]
I found the source of the issue for PostCSS config and submited a pull request:
https://github.com/laravel-frontend-presets/tall/pull/128

The __dirname error is something to do with VSCode and the remote linux server. Im lost on this one.

Thanks for reading my rants, this has nothing to do with Tailwind CSS.

**Comment by alirezamajdi** [comment]
Rename file name postcss.config.js to postcss.config.cjs

**Reply by CallMeProteus** [reply]
Literary written in the log error🤦‍♂️
